kernel/packages/unity-interface/realmsForRenderer.ts

```diff
@@ -0,0 +1,60 @@
+import { StoreContainer } from '../shared/store/rootTypes'
+import { getExploreRealmsService, getRealm } from '../shared/dao/selectors'
+import { RealmsInfoForRenderer } from '../shared/types'
+import { observeRealmChange } from 'shared/dao'
+import { Realm } from 'shared/dao/types'
+import { unityInterface } from './UnityInterface'
+
+const REPORT_INTERVAL = 2 * 60 * 1000
+
+declare const globalThis: StoreContainer
+
+let isReporting = false
+
+export function startRealmsReportToRenderer() {
+  if (!isReporting) {
+    isReporting = true
+
+    const realm = getRealm(globalThis.globalStore.getState())
+    if (realm) {
+      reportToRenderer({ current: convertRealmType(realm) })
+    }
+
+    observeRealmChange(globalThis.globalStore, (previous, current) => {
+      reportToRenderer({ current: convertRealmType(current) })
+    })
+
+    fetchAndReportRealmsInfo()
+
+    reportInterval()
+  }
+}
+
+function reportInterval() {
+  setInterval(() => {
+    fetchAndReportRealmsInfo()
+    reportInterval()
```

Comment:
If you use `setInterval` you don't need to loop. It is done automatically. But I'd suggest replacing the `setInterval` for a `setTimeout`

Reply:
upsi... I was thinking `setTimeout` and typed `setInterval` LOL Thx